src/main/java/org/apache/sysds/runtime/controlprogram/federated/FederatedWorkerHandler.java

```diff
@@ -268,7 +268,8 @@ private FederatedResponse execUDF(FederatedRequest request) {
 			return udf.execute(ec, inputs);
 		}
 		catch(Exception ex) {
-			return new FederatedResponse(ResponseType.ERROR, ex.getMessage());
+			return new FederatedResponse(ResponseType.ERROR, new FederatedWorkerHandlerException(
+				"Exception of type " + ex.getClass() + " thrown when processing request", ex));
```

Comment:
`ex.getMessage()` would sometimes not suffice and instead be confusing. A `ArrayIndexOutOfBoundsException` would have for example just have the index as a message, therefore our response would just contain an arbitrary number and the problem would not be explained